test/hexpm/web/views/package_view_test.exs

```diff
@@ -107,4 +107,51 @@ defmodule Hexpm.Web.PackageViewTest do
     assert PackageView.snippet_version(:erlang_mk, version3) == "2.0.2"
   end
 
+  describe "retirement_message/1" do
+    test "reason is 'other', message contains text" do
+      retirement = %{reason: "other", message: "something went terribly wrong"}
+      assert PackageView.retirement_message(retirement) == ["Retired package", "; reason: ", "something went terribly wrong"]
+    end
+
+    test "reason is 'other', message is empty" do
+      retirement = %{reason: "other", message: nil}
+      assert PackageView.retirement_message(retirement) == ["Retired package"]
+    end
+
+    test "reason is not 'other', message contains text" do
+      retirement = %{reason: "security", message: "something went terribly wrong"}
+      assert PackageView.retirement_message(retirement) == ["Retired package", "; reason: ", "Security issue", " - ", "something went terribly wrong"]
+    end
+
+    test "reason is not 'other', message is empty" do
+      retirement = %{reason: "security", message: nil}
+      assert PackageView.retirement_message(retirement) == ["Retired package", "; reason: ", "Security issue"]
+    end
+  end
+
+  describe "retirement_html/1" do
+    test "reason is 'other', message contains text" do
+      retirement = PackageView.retirement_html(%{reason: "other", message: "something went terribly wrong"})
+      message = Enum.map_join(retirement, fn(x) -> if is_tuple(x), do: Phoenix.HTML.safe_to_string(x), else: x end)
+      assert message == "<strong>Retired package;</strong> reason: something went terribly wrong"
```

Comment:
This should also work:

```elixir
retirement = PackageView.retirement_html(%{reason: "other", message: "something went terribly wrong"})
assert Phoenix.HTML.safe_to_string(retirement) == "<strong>Retired package;</strong> reason: something went terribly wrong"
```

Reply:
in that case, retirement actually returns something like this: `[{:safe, [60, "strong", [], 62, "Retired package;", 60, 47, "strong", 62]}, " reason: ", "something went terribly wrong"]`

`Phoenix.HTML.safe_to_string` throws an error because it is expecting a tuple with the first element being the atom `:safe`. This code passes an array and it throws an exception. That is why i m doing the map, to apply the function to the tuple(s) and ignore the strings.

perhaps i have an older/newer version of phx installed that acts differently than what you expect.

Reply:
Here is the exact output from an IEx
```
iex(1)> retirement = [{:safe, [60, "strong", [], 62, "Retired package;", 60, 47, "strong", 62]}, " reason: ", "something went terribly wrong"]
[{:safe, [60, "strong", [], 62, "Retired package;", 60, 47, "strong", 62]},
 " reason: ", "something went terribly wrong"]
iex(2)> Phoenix.HTML.safe_to_string(retirement)
** (FunctionClauseError) no function clause matching in Phoenix.HTML.safe_to_string/1    
    
    The following arguments were given to Phoenix.HTML.safe_to_string/1:
    
        # 1
        [{:safe, [60, "strong", [], 62, "Retired package;", 60, 47, "strong", 62]},
     " reason: ", "something went terribly wrong"]
    
    Attempted function clauses (showing 1 out of 1):
    
        def safe_to_string({:safe, iodata})
    
    (phoenix_html) lib/phoenix_html.ex:160: Phoenix.HTML.safe_to_string/1
```

Reply:
Okay, I assumed `safe_to_string/1` accepted anything that was accepted in safe templates, sorry about that.

We can create a local function in the test module that does the `map_join` you have here and call that in all tests.